### DISCUSSION TITLE
Bad municipality from reverse geocoding search in enclaved city

### DISCUSSION THREAD

**Discussion by wolfy1339** [osm-search/Nominatim#3508] [Geocoding around the world]
I am seeing some invalid results for the geocoding,
https://nominatim.openstreetmap.org/ui/details.html?osmtype=W&osmid=849626224&class=leisure

It seems that it is catching on to the neighbourhood of the city next door (Saint-Émile; [r8387702](https://www.openstreetmap.org/relation/8387702)) instead of the proper city (Wendake; [r7701011](https://www.openstreetmap.org/relation/7701011)).

The city is an enclave within another, so that doesn't help...
The admin boundary relations all seem correct, when you use the Query features option on the OSM main page, it doesn't display the irrelevent admin boundaries.

https://www.openstreetmap.org/query?lat=46.87078&lon=-71.35158

Any input on wether this is a bug or intended behaviour?

**Comment by lonvia** [comment]
This looks like a bug. My guess is that there is no admin_level=10 boundary in this area, so it just takes the closest. It shouldn't do that when it is an area. Needs confirming.

**Comment by lonvia** [comment]
So, Quebec and Saint-Emile are in there because the [parent street](https://www.openstreetmap.org/way/849626221) goes into the administrative area of Quebec. This is sadly one of the corner cases that are very expensive to fix. Nominatim would be able to "fix" the address assignment on-the-fly when there is another level 10 boundary to replace the wrong Saint-Emile with. But as Wendake doesn't seem to have level 10 boundaries, it can't completely remove them without making all address computations more expensive.
